### PR TITLE
Show initialization error messages only if SCOPE_DEBUG is set

### PR DESCRIPTION
**Pull request by tonyredondo**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -237,7 +237,7 @@ func NewAgent(options ...Option) (*Agent, error) {
 		} else {
 			agent.logger.Println("API key not found, agent can't be started")
 			return nil, errors.New(fmt.Sprintf("There was a problem initializing Scope.\n"+
-				"Check the agent logs at %s for more information.\n", agent.recorderFilename))
+				"Check the agent logs at %s for more information.\n\nScope DSN not found, agent can't be started", agent.recorderFilename))
```
Comment (drodriguezhdez) [inline agent/agent.go:240]: I think message was changed to:
```
SCOPE_DSN not defined, test results will not be sent to Scope.
Developing locally? Try out Scratchpad! https://app.scope.dev/local-dev/
```

File: init.go [hunk 2]
```diff
@@ -34,7 +35,9 @@ func Run(m *testing.M, opts ...agent.Option) int {
 	newAgent, err := agent.NewAgent(opts...)
 	if err != nil {
 		res := m.Run()
-		fmt.Printf("\n%v\n", err)
+		if env.ScopeDebug.Value {
```
Comment (drodriguezhdez) [inline init.go:38]: Is this line preventing to show the message if `SCOPE_DSN` is not set? (previous message)

File: agent/agent.go (hunk 1 above)
Comment (tonyredondo) [reply agent/agent.go:240]: Yeah but that message was when we discuss to show it always, according to @fermayo now this message will show only if the debug flag is true. So the marketing thing of the second line makes no sense to me.

File: init.go (hunk 2 above)
Comment (tonyredondo) [reply init.go:38]: This line is preventing to show any initialization message if the debug flag is not set.

File: agent/agent.go (hunk 1 above)
Comment (drodriguezhdez) [reply agent/agent.go:240]: OK, then we need to synchronize each other with this topic. 
Java agent does not have even `debug` flag. It has logger levels.

Comment (fermayo) [reply agent/agent.go:240]: Can we change the full message to just:
```
Scope DSN not found. Tests will run but no results will be reported to Scope. More info at https://docs.scope.dev
```
when debug is enabled.

Comment (tonyredondo) [reply agent/agent.go]: done

**Pull request merged**